client/src/app/queries/assessments.ts

```diff
@@ -210,3 +216,45 @@ const removeSectionOrderFromQuestions = (
     })),
   };
 };
+
+export const useFetchAssessmentsWithArchetypeApplications = () => {
+  const { assessments, isFetching: assessmentsLoading } = useFetchAssessments();
+
+  const archetypeQueries = useQueries({
+    queries:
+      assessments?.map((assessment) => ({
+        queryKey: ["archetype", assessment.archetype?.id],
+        queryFn: () =>
+          assessment.archetype?.id
+            ? getArchetypeById(assessment.archetype.id)
+            : undefined,
+        enabled: !!assessment.archetype?.id,
+      })) || [],
+  });
+
+  const isArchetypesLoading = archetypeQueries.some((query) => query.isLoading);
+
+  const archetypeApplicationsMap = new Map();
+  archetypeQueries.forEach((query, index) => {
+    if (query.data && assessments[index].archetype?.id) {
+      archetypeApplicationsMap.set(
+        assessments[index]?.archetype?.id,
+        query.data.applications
+      );
+    }
+  });
+
+  const assessmentsWithArchetypeApplications = assessments.map(
+    (assessment) => ({
+      ...assessment,
+      archetypeApplications: assessment.archetype?.id
+        ? archetypeApplicationsMap.get(assessment.archetype.id) || []
```

Comment:
`.get()` should work even if given `undefined` and just return `undefined` so could just be:
```js
    archetypeApplications: archetypeApplicationsMap.get(assessment.archetype.id) ?? [],
```